From: Daniel van Vugt <daniel.van.vugt@canonical.com>
Date: Tue, 19 Sep 2023 17:39:07 +0800
Subject: kms/impl-device: Inhibit deadline timer on amdgpu

Recent AMD Ryzen systems are experiencing a problem whereby gnome-shell
gets SIGKILLed by the kernel. It seems to be related to a violation of
RLIMIT_RTTIME in the KMS thread.

This is the simplest/safest workaround we have right now. Longer term
we would like to:

1. Find out what call is blocking in amdgpu to trigger it; and
2. Graceful handling of SIGXCPU before SIGKILL happens.

https://launchpad.net/bugs/2034619
https://gitlab.freedesktop.org/drm/amd/-/issues/2861

Closes: https://gitlab.gnome.org/GNOME/mutter/-/issues/3037

Forwarded: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3291
---
 src/backends/native/meta-kms-impl-device.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/backends/native/meta-kms-impl-device.c b/src/backends/native/meta-kms-impl-device.c
index 972873a..56d493b 100644
--- a/src/backends/native/meta-kms-impl-device.c
+++ b/src/backends/native/meta-kms-impl-device.c
@@ -1902,6 +1902,7 @@ maybe_inhibit_deadline_timer (MetaKmsImplDevice *impl_device)
   MetaKmsImplDevicePrivate *priv =
     meta_kms_impl_device_get_instance_private (impl_device);
   static const char *deadline_timer_deny_list[] = {
+    "amdgpu",
     "vc4",
   };
   int i;
